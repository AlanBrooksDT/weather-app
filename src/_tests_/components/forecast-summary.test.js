import React from 'react';
import { render } from "@testing-library/react";
import ForecastSummary from '../../components/forecast-summary';

describe("ForecastSummary", () => {

    it('renders', () => {
        const { asFragment } = render(
            <ForecastSummary
                date={1525046400000}
                temperature={30}
                description='testDesc'
                icon='800'
            />
        )

        expect(asFragment()).toMatchSnapshot();
    })

    it('renders a number of props into a weather summary', () => {
        const { getByText, getByTestId } = render(
            <ForecastSummary
                date={1525046400000}
                temperature={9}
                description='Clear'
                icon='800'
            />
        )

        expect(getByText('April 30th, 2018')).toBeTruthy();
        expect(getByText('April 30th, 2018')).toHaveClass('forecast-summary-date');
        expect(getByTestId('temperature-id')).toHaveClass('temperature');
        expect(getByText('Clear')).toHaveClass('description');
    })
})