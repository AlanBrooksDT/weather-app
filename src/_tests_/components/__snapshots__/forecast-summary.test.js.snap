// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
  >
    <span
      class="date"
    >
      mockDate
    </span>
    <span
      class="temperature"
    >
       mockTemperature
    </span>
    <span
      class="description"
    >
      mockDescription
    </span>
    <span
      class="icon"
    >
      mockIcon
    </span>
  </div>
</DocumentFragment>
`;
